zilliqa/tests/it/main.rs

```diff
@@ -253,7 +254,7 @@ impl Network {
                     k.node_public_key(),
                     k.to_libp2p_keypair().public().to_peer_id(),
                     stake.into(),
-                    Address::random_with(rng.lock().unwrap().deref_mut()),
+                    Address::from_private_key(&k.as_ecdsa()),
```

Comment:
Why this change? I guess this relates to my previous point.

Reply:
Oh I see - Because we need to send the removal transaction from the `rewardAddress` account. I'd slightly prefer for us to just generate a distinct ECDSA key for each `TestNode` and expose a `Wallet` from there, since it keeps the two keys distinct.

But I won't block if you think this is okay :)

Reply:
That's very fair - I think I'll do it this way then. In the docker compose, the default config re-uses the private keys for the reward addresses so I kind of operated on the assumption that each node has a "natural" address, but I'm perfectly happy to decouple them.

However I think I'll do it in a separate PR, alongside cleaning up the ECDSA methods for SecretKey then.